test/router/path_trie_tail_test.dart

```diff
@@ -0,0 +1,265 @@
+import 'package:relic/src/router/normalized_path.dart';
+import 'package:relic/src/router/path_trie.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('PathTrie Tail (**) Matching', () {
+    late PathTrie<int> trie;
+
+    setUp(() {
+      trie = PathTrie<int>();
+    });
+
+    test(
+        'Given a trie with path /static/**, '
+        'when /static/css/style.css is looked up, '
+        'then it matches with correct value and remaining path', () {
+      trie.add(NormalizedPath('/static/**'), 1);
+      final result = trie.lookup(NormalizedPath('/static/css/style.css'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/static');
+      expect(result.remaining.path, '/css/style.css');
+    });
+
+    test(
```

Comment:
Nit: This is basically the same as above, just with more "sub folders", right?

At first I was looking for another different as the switch from `static` to `files` etc., but I don't think there is one.

Reply:
Not sure I follow. This is the very first test in the file.

Reply:
It is the second one, and basically (I think) just a variant / enhancement of the first, but since it was using different names for the use-case under test (`/files` instead of `/static`) I at first thought they were more different than they turned out to be.

But no big deal, just that I think that having the same setup could help seeing that's actually under test (and thus always using the same "example", or just `/a/b/c.d` consistently).

Reply:
I have gone over the tests and removed "duplicates"